lib/port_scanner.py

```diff
@@ -0,0 +1,45 @@
+import queue
+import socket
+from threading import Thread, Lock
+from queue import Queue
+
+N_THREADS = 150
+
+queue = Queue()
+print_lock = Lock()
+
+def scan_port(host, port):
+    try:
+        sock = socket.socket()
+        sock.connect((host, port))
+    except:
+        pass
+    else:
+        with print_lock:
```

Comment:
This else + with block is unnecesarry.
```python
    try:
        sock = socket.socket()
        sock.connect((host, port))
        print(f"{host:15}:{port:5} is open")
    except:
        pass
    finally:
        sock.close()
```

Reply:
Why do we need those Locks at all?

Reply:
I used it because it makes that only one thread can print at a time but I didn't encountered any difference in how outputs are presented for now

Reply:
I'll delete them for now and eventually get back it we encounter any problems as I mentioned

Reply:
We would need to use locks when the threads would modify data. Then we would need to make sure only one thread is doing that at once. Writing to stdout is not one of those cases I don't think.